lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -354,6 +354,8 @@ abstract class BaseSlider<
   private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener =
       this::updateLabels;
 
+  private boolean isVisible = true;
```

Comment:
@pubiqq just wondering about the default value being true here. if the Slider visibility is set to `gone` or `invisible` initially (e.g. in the xml layout), will onVisibilityAggregated() be called with false immediately after the view is inflated / created?

if not then maybe in the constructor we could set `isVisible = getVisibility() == View.VISIBLE`?

Reply:
I just checked locally and `onVisibilityAggregated` doesn't seem to get called when the Slider itself has `android:visibility="invisible"` or `android:visibility="gone"` in the XML layout.

So the `isVisible` boolean would be incorrect in those cases, which makes me think we should not initialize `isVisible` to true and instead do `isVisible = getVisibility() == View.VISIBLE` in the constructor.

Although that doesn't cover the case where the Slider is `visible` but the parent view is `gone`. because in that case `getVisibility() == View.VISIBLE` would return true in the Slider constructor and `onVisibilityAggregated` doesn't get called after the initial inflation to correct it.

Reply:
Thank you very much. Following pubiqq's advice, I replaced the isVisible field with the isShown() method. I tried displaying labels in the SliderScrollContainerDemoFragment demo. There seems to be no problem.
https://github.com/material-components/material-components-android/blob/master/catalog/java/io/material/catalog/slider/SliderScrollContainerDemoFragment.java
```
slider.setLabelBehavior(LabelFormatter.LABEL_VISIBLE);